agent/eventhandler/task_handler_types_test.go

```diff
@@ -153,11 +153,21 @@ func TestShouldTaskEventBeSent(t *testing.T) {
 				},
 				ManagedAgents: []api.ManagedAgentStateChange{
 					{
-						TaskArn:       "test_task_arn",
-						Name:          "test_agent",
-						ContainerName: "test_container",
-						Status:        apicontainerstatus.ManagedAgentStatusNone,
-						Reason:        "test_reason",
+						TaskArn: "test_task_arn",
+						Name:    "test_agent",
+						Container: &apicontainer.Container{
+							ManagedAgentsUnsafe: []apicontainer.ManagedAgent{
+								{
+									Name: "test_agent",
+									ManagedAgentState: apicontainer.ManagedAgentState{
+										Status:     apicontainerstatus.ManagedAgentRunning,
+										SentStatus: apicontainerstatus.ManagedAgentStatusNone,
+									},
```

Comment:
could we also add a test case where managedAgent sent >= known state and check if task event is not sent?

Reply:
this actually brought up a good point that I'd missed: 
it shouldn't be `>=` for managed agent, it should be `!=` as the state can shift from running -> stopped -> running.  Updating.